poi-ooxml/src/main/java/org/apache/poi/openxml4j/util/ZipSecureFile.java

```diff
@@ -129,13 +135,22 @@ public static long getMaxTextSize() {
     }
 
     public ZipSecureFile(File file) throws IOException {
-        super(file);
-        this.fileName = file.getAbsolutePath();
+        this(file.toPath());
+    }
+
+    public ZipSecureFile(Path path) throws IOException {
+        super(
+            Files.newByteChannel(path, EnumSet.of(StandardOpenOption.READ)),
```

Comment:
why do we open the file for only read access here?

Reply:
javadoc for the `super` implementation (`ZipFile`):

```
Opens the given channel for reading, assuming the specified encoding for file names.
org.apache.commons.compress.utils.SeekableInMemoryByteChannel allows you to read from an in-memory archive.

Reply:
Okay then can we just have a comment like `\\ Opens a new byte channel for reading per the Super implementation of ZipFile. between lines 142 and 143?`

Reply:
I want to make it very clear that it's good insight to question things like this. Being explicit about the operations performed is not a bad thing in and of itself. 

Where I disagree with the need for something like this is that `ZipFile` is a file defined within this same project. While the need for this might not be immediately intuitive, it takes no more than reading the docs of the used parent constructor to determine that we're doing this for compliance with the API we're calling.

To me, this is akin to commenting our call to some Crawler in Unpacker with something like

```
crawl(
    context,
    file.toPath(), // Convert to Path per the implementation of [DocumentCrawler]
)
```

Again, it's not like it's bad to be explicit in certain scenarios, but IMO, anyone curious about why this is read only could easily  Having frequent comments of a somewhat trivial nature throw off the signal-to-noise ratio. I'll happily adjust if people disagree with what I'm saying.

Reply:
it's just not intuitive to me reading through this code as to why only READ permission is warranted here and I shouldnt have to fire up an IDE just to go/reference the super implementation is all

Reply:
@PKJosh - is `READ` access an _engineering choice_ you made, or is literally just matching the super / documentation?
If the former, I'd lean towards documenting the decision with a comment.
If the latter, I'd lean towards saying it's already documented.

Reply:
It's the latter. It was merely compliance with the existing API.

Effectively, `// I did what was expected of me` 😛